### PR TITLE
Docs: Add a note about Loki labels format

### LINKED ISSUE #9928: Fix docs: promtail configuration: mention the correct use of <labelname>
**Describe the bug**

In Promtail configuration, when using labels, it's not clearly mentioned that the label name must be a valid regular expression `[a-zA-Z_][a-zA-Z0-9_]*`. And that any special character needs to be converted to an underscore.

See a probably related bug: https://github.com/grafana/loki/issues/9113
See this bug + fix in the Prometheus project: https://github.com/prometheus/docs/issues/735

**To Reproduce**
Try using the following scrape config:

```
scrape_configs:

- job_name: docker

  docker_sd_configs:
    - host: unix:///var/run/docker.sock
  
  # See: https://grafana.com/docs/loki/latest/clients/promtail/configuration/
  relabel_configs:
    - source_labels: ['__meta_docker_container_id']
      regex: '(.*)'
      target_label: 'container_id'
    - source_labels: ['__meta_docker_container_name']
      regex: '/(.*)'
      target_label: 'container_name'
    - source_labels: ['__meta_docker_container_log_stream']
      regex: '(.*)'
      target_label: 'container_log_stream'
    - source_labels: ['__meta_docker_network_ip']
      regex: '(.*)'
      target_label: 'container_ip'
    - source_labels: ['__meta_docker_container_label_com.docker.swarm.service.name']
      regex: '/(.*)'
      target_label: 'app'
```

**Expected behavior**
The docker label should be taken and injected into Loki.
But you get a runtime error when trying to start Promtail, telling you that the configuration is invalid. 

The solution is to convert any special character (dashes, dot, slashes) to underscores, and specifically here to: `__meta_docker_container_label_com_docker_swarm_service_name`

**Environment:**
Docker Swarm, but the same is probably true for Kubernetes, and every other place that "\<labelname\>" is being used.

**Screenshots, Promtail config, or terminal output**
See above.

### PR DESCRIPTION
**Pull request by edsoncelio**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #9928

**Special notes for your reviewer**:
This PR adds a note about an important requirement in the format of Loki labels (with a small example).

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/get-started/labels/_index.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ Loki places the same restrictions on label naming as [Prometheus](https://promet
 >
 > Note: The colons are reserved for user defined recording rules. They should not be used by exporters or direct instrumentation.
 
+{{% admonition type="note" %}}
+Any other unsupported character in the label should be converted to an underscore. For example, the label `app.kubernetes.io/name` should be written as `app_kubernetes_io_name`
```
Comment (JoaoBraveCoding) [inline docs/sources/get-started/labels/_index.md:36]: nit
```suggestion
Unsupported characters in the label should be converted to an underscore. For example, the label `app.kubernetes.io/name` should be written as `app_kubernetes_io_name`
```

**Review by JoaoBraveCoding — COMMENTED** [review]
Minor nit, but LGTM according to the discussion on the issue.

Disclaimer: I'm trying to help out with PR reviews, and learn the project. I'm not an official maintainer/reviewer of the project.

File: docs/sources/get-started/labels/_index.md (hunk 1 above)
Comment (edsoncelio) [reply docs/sources/get-started/labels/_index.md]: Sounds good, thanks for the suggestion @JoaoBraveCoding :)

**Review by JStickler — APPROVED** [review]
[docs team]  LGTM

**Pull request merged**